### PR TITLE
Sort lessons index

### PR DESCRIPTION
**Pull request by cmkoller**
Lessons default sorted by title
Added filters for sorting lessons by type

### CODE REVIEW

File: app/views/lessons/index.html.haml [hunk 1]
```diff
@@ -6,6 +6,18 @@
 
 %h2 Lessons
 
+%dl.sub-nav
+  %dt Filter:
```
Comment (dpickett) [inline app/views/lessons/index.html.haml]: I'm fairly certain for HTML to validate that a `dt` must have one and only one `dd`. I'd consider an unordered list here instead

Comment (dpickett) [inline app/views/lessons/index.html.haml]: I'd love to see this in a view object. (Richard or I can help with that) - I think there's a way to leverage an array to DRY this up a bit.

File: spec/features/filter_lessons_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  scenario "filter lessons to only view tutorial" do
+    visit lessons_path
+    click_link "Tutorials"
+
+    expect(page).to have_link(tutorial.title)
+    expect(page).to have_no_link(article.title)
+    expect(page).to have_no_link(exercise.title)
+    expect(page).to have_no_link(challenge.title)
```
Comment (dpickett) [inline spec/features/filter_lessons_spec.rb]: excellent and thorough acceptance test!

File: app/views/lessons/index.html.haml (hunk 1 above)
Comment (cmkoller) [inline app/views/lessons/index.html.haml]: I tried validating the code and didn't get any errors on the dt/dd thing! I originally decided to use them in order to take advantage of Foundation's [sub nav styling](http://foundation.zurb.com/docs/components/subnav.html), but if you'd prefer them changed I can probably just replicate that styling myself on a `ul`! Let me know what you think is best.

I'd love to get input and maybe pair with you or Richard as to making this into a view object - I don't think that's something I've done before! I'll see if Richard has time to help me with that.

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class LessonsController < ApplicationController
   def index
-    @lessons = filter_lessons(Lesson.order(:position))
+    @lessons = filter_lessons(ordered_lessons)
```
Comment (atsheehan) [inline app/controllers/lessons_controller.rb]: Nice use extracting this into a separate method.

**Pull request merged**